docs/src/jekyll/docs.md

```diff
@@ -0,0 +1,546 @@
+case class PostInfo(topic: String)
+```
+
+As you can see, every `Post` has an author, but it refers to it by its id. We'll implement two data sources:
+
+- one for retrieving a post given a post id
+- another for retrieveng post metadata given a post id
+
+```scala
+implicit object PostSource extends DataSource[PostId, Post]{
+  override def fetch(ids: List[PostId]): Eval[Map[PostId, Post]] = {
+    val result = ids.map(i => (i, Post(i, i % 3 + 1, s"An article with id $i"))).toMap
+    latency(result, s"Fetching posts $ids")
+  }
+}
+
+def getPost(id: PostId): Fetch[Post] = Fetch(id)
+
+implicit object PostInfoSource extends DataSource[PostId, PostInfo]{
+  override def fetch(ids: List[PostId]): Eval[Map[PostId, PostInfo]] = {
+    val result = ids.map(i => (i, PostInfo(if (i % 2 == 0) "applicative" else "monad"))).toMap
+    latency(result, s"Fetching post info $ids")
+  }
+}
+  
+def getPostInfo(id: PostId): Fetch[PostInfo] = Fetch(id)
+```
+
+We can also implement a function for fetching a post's author given a post:
+
+```scala
+def getAuthor(p: Post): Fetch[User] = Fetch(p.author)
+```
+
+Now that we have multiple sources, let's mix them in the same fetch.
+
+```scala
+val fch: Fetch[(Post, User)] = for {
+  post <- getPost(1)
+  user <- getAuthor(post)
+} yield (post, user)
+
+val fut: Future[(Post, User)] = Fetch.run(fch)
+
+await(fut)
+// ~~~>[thread: 81] Fetching posts List(1)
+// <~~~[thread: 81] Fetching posts List(1)
+// ~~~>[thread: 82] Fetching users List(2)
+// <~~~[thread: 82] Fetching users List(2)
+
+//=> (Post(1,2,An article with id 1),User(2,@egg_2))
+```
+
+In the previous example we fetched a post given its id, and then fetched its author. These
+data could come from entirely different places but Fetch makes working with heterogeneous sources
+of data very easy.
+
+### Parallelism
+
+Combining multiple independent requests to the same data source results in batching but what happens
+when combining independent requests to *different* data sources? We have the oportunity to query the
+different data sources at the same time.
+
+Since we are using `Future` as our concurrency monad, the requests will effectively run in parallel,
+each in its own thread.
+
+```scala
+val fch: Fetch[(Post, User)] = Fetch.join(getPost(1), getUser(2))
+val fut: Future[(Post, User)] = Fetch.run(fch)
+
+await(fut)
+// ~~~>[thread: 83] Fetching posts List(1)
+// ~~~>[thread: 84] Fetching users List(2)
+// <~~~[thread: 84] Fetching users List(2)
+// <~~~[thread: 83] Fetching posts List(1)
+
+//=> (Post(1,2,An article with id 1),User(2,@egg_2))
+```
+
+As you can notice, both requests are started at the same time in different threads (with ids 83 and 84) and
+the fetch finishes as soon as both results are available.
+
+## Combinators
+
+Besides `flatMap` for sequencing fetches and `join` for running them concurrently, Fetch provides a number of
+other combinators that you can use.
+
+### Collect
```

Comment:
Is order respected in this combinator? Also collect may be a misleading name because of the existing `collect` combinator for `map + filter` using Partial functions predicates